### PR TITLE
Fix docker volumes

### PR DESCRIPTION
**Pull request by Poltergeist**
This also fixes the problem of adding a new row at the top of the form if no row was present.
 
image

image

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @philipnrmn

File: src/js/utils/ServiceUtil.js [hunk 1]
```diff
@@ -123,6 +123,9 @@ const ServiceUtil = {
           if (volumes.dockerVolumes) {
             volumesList = volumesList.concat(
               volumes.dockerVolumes
+                .filter(function ({containerPath, hostPath}) {
```
Comment (kennyt) [inline src/js/utils/ServiceUtil.js]: should this be indented at the same level as `map` below?

**Comment by kennyt** [comment]
not 100% how this works, but if i fill all the fields out, switch to JSON then switch back container volumes is empty

px

File: src/js/components/SchemaForm.js [hunk 2]
```diff
@@ -136,10 +136,10 @@ class SchemaForm extends mixin(StoreMixin) {
     }
 
     return (
-      <div className="form-row-element">
+      <div className="form-row-element"
+        prop={prop}>
```
Comment (kennyt) [inline src/js/components/SchemaForm.js]: can you say something about what this is used for? not 100% clear to me atm

**Comment by rcorral** [comment]
Guys, I'm going to merge this in so that we can provide a build. But @Poltergeist could you please take a look at the comments Kenny left?

**Comment by kennyt** [comment]
sounds good

**Comment by rcorral** [comment]
integration tests pass.

File: src/js/components/SchemaForm.js (hunk 2 above)
Comment (Poltergeist) [inline src/js/components/SchemaForm.js]: This is needed to find the exact place to add a new row. If this is not present the fields get added at the beginning of the form rather then above the button. This is only needed in the case if there isn't any row left.

**Comment by Poltergeist** [comment]
The docker volume gets cleared if no docker image is present. Ideal would be to only show those fields if a docker volume is set.

**Pull request merged**